modules/dnn/CMakeLists.txt

```diff
@@ -28,7 +28,7 @@ if(MSVC)
   )
 else()
   ocv_warnings_disable(CMAKE_CXX_FLAGS -Wno-deprecated -Wmissing-prototypes -Wmissing-declarations -Wshadow
-                                       -Wunused-parameter -Wsign-compare
+                                       -Wunused-parameter -Wsign-compare -Wundef
```

Comment:
`-Wundef` is very important warning (no other way to detect typos in `#if` statements), so please minimize scope of this effect (put under `if(HAVE_CUDA)`)

Reply:
Thank you !
I understand your request, and updated the CMakeLists.txt